in_toto/in_toto_verify.py

```diff
@@ -45,104 +70,34 @@
-  """
-  <Purpose>
-    Loads the layout metadata as Metablock object (containing a Layout object)
-    and the signature verification keys from the passed paths and/or from
-    layout_gpg_keyids, calls verifylib.in_toto_verify
-    and handles exceptions.
-
-    The layout has to be signed by the private key corresponding to each passed
-    public key (path) or gpg key (keyid). If any of the signatures are missing
-    or invalid verification fails.
-
-  <Arguments>
-    layout_path:
-            Path to layout metadata file that is being verified.
-
-    layout_key_paths:
-            List of path(s) to project owner public key(s), used to verify the
-            root layout's signature(s).
-
-    layout_gpg_keyids:
-            List of project owner GPG keyid(s), used to verify the root
-            layout's signature(s).
-
-    gpg_home:
-            Path to GPG keyring (if not set the default keyring is used).
-
-  <Exceptions>
-    SystemExit if any exception occurs
-
-  <Side Effects>
-    Calls sys.exit(1) if an exception is raised
-
-  <Returns>
-    None.
-
-  """
-  try:
-    log.info("Verifying software supply chain...")
-
-    log.info("Loading layout...")
-    layout = Metablock.load(layout_path)
-
-    layout_key_dict = {}
-    if layout_key_paths != None:
-      log.info("Loading layout key(s)...")
-      layout_key_dict.update(
-          in_toto.util.import_rsa_public_keys_from_files_as_dict(
-          layout_key_paths))
-
-    if layout_gpg_keyids != None:
-      log.info("Loading layout gpg key(s)...")
-      layout_key_dict.update(
-          in_toto.util.import_gpg_public_keys_from_keyring_as_dict(
-          layout_gpg_keyids, gpg_home=gpg_home))
-
-    verifylib.in_toto_verify(layout, layout_key_dict)
-
-  except Exception as e:
-    log.info("{0} - {1}".format(type(e).__name__, e))
-    sys.exit(1)
 
 def main():
   """Parse arguments and call in_toto_verify. """
   parser = argparse.ArgumentParser(
       description="Verifies in-toto final product")
 
-  # Whitespace padding to align with program name
-  lpad = (len(parser.prog) + 1) * " "
+  parser.usage = "%(prog)s <named arguments> [optional arguments]"
 
-  parser.usage = ("\n"
-      "%(prog)s --layout <layout path>\n{0}"
-               "{{--layout-keys <filepath>[ <filepath> ...], "
-               " --gpg <keyid> [ <keyid> ...]}} \n{0}"
-               "[--gpg-home <path to gpg keyring>]\n{0}"
-               "[--verbose | --quiet]\n\n"
-               .format(lpad))
+  named_args = parser.add_argument_group("required named arguments")
 
-  in_toto_args = parser.add_argument_group("in-toto options")
+  named_args.add_argument("-l", "--layout", type=str, required=True,
+      help="Root layout to use for verification", metavar="<layout path>")
 
-  in_toto_args.add_argument("-l", "--layout", type=str, required=True,
-      help="Root layout to use for verification")
+  named_args.add_argument("-k", "--layout-keys", type=str,
```

Comment:
This is due to a shortcoming of `argparse`. There's no way of specifying a required (non mutual exclusive) group. If I'd make each argument required, the user couldn't specify only one of them, which is what I want.
I'll update the help message so that it's clear that only one of layout keys or gpg ids is okay.